modules/geo-layers/src/dggs-layer/ddgs-layer.ts

```diff
@@ -0,0 +1,50 @@
+// deck.gl
+// SPDX-License-Identifier: MIT
+// Copyright (c) vis.gl contributors
+
+import {AccessorFunction, DefaultProps} from '@deck.gl/core';
+import GeoCellLayer, {GeoCellLayerProps} from '../geo-cell-layer/GeoCellLayer';
+import {DGGSDecoder} from './dggs-decoders/dggs-decoder';
+import {flattenPolygon} from '../h3-layers/h3-utils';
+
+/** All properties supported by DGGSLayer. */
+export type DGGSLayerProps<DataT = unknown> = _DGGSLayerProps<DataT> & GeoCellLayerProps<DataT>;
+
+/** Properties added by DGGSLayer. */
+type _DGGSLayerProps<DataT> = {
+  /** Called for each data object to retrieve the DGGS cell identifier. By default, it reads `cellId` property of data object. */
+  getCellId?: AccessorFunction<DataT, string | bigint>;
+  /** The DGGS decoder to use. */
+  dggsDecoder: DGGSDecoder;
+};
+
+/** Render filled and/or stroked polygons based on the specified DGGS geospatial indexing system. */
+export class DGGSLayer<DataT = any, ExtraProps extends {} = {}> extends GeoCellLayer<
+  DataT,
+  Required<_DGGSLayerProps<DataT>> & ExtraProps
+> {
+  static layerName = 'DGGSLayer';
+  static defaultProps: DefaultProps<DGGSLayerProps> = {
+    getCellId: {type: 'accessor', value: (d: any) => d.cellId},
+    dggsDecoder: {type: 'object', compare: true, value: undefined!}
```

Comment:
How about following the pattern we have in `TileLayer` where `TilesetClass` is used as a prop?

Reply:
Well... that particular prop happens to be one of my pet peeves with deck.gl 😄 

(In my opinion one should pass on a Tileset **instance** that could be shared by multiple Tilelayers)


Other than that, I just prefer simple types/interfaces and plain objects in favor of classes.

But if this is a sticking point I could reconsider.